test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -552,14 +552,16 @@ describe('Client Side Encryption Prose Tests', function () {
     const limitsKey = loadLimits('limits-key.json');
     const limitsDoc = loadLimits('limits-doc.json');
 
-    before(function () {
-      // First, perform the setup.
+    let firstTimeSetup = true;
+    beforeEach(async function () {
+      if (firstTimeSetup) {
+        firstTimeSetup = false;
```

Comment:
So `before` hooks will still run if tests are skipped which was causing an issue here, the before hook was doing logic that also depended on a certain mongo version. So introducing this change would add a kind of hazzard. This is similar to how our session leak checker is the first afterEach that runs so you can't use afterEach to close clients currently. 
I still think the change is worth it and we can follow up with some better work around potentially later. Thoughts? 

I will add notes about both of these special scenarios to the test/readme